profiles/views.py

```diff
@@ -55,3 +56,36 @@ def get_serializer_class(self):
             return self.serializer_class_limited
         # Case #5: this should never happen, but just in case
         return self.serializer_class_private
+
+    def get_object(self):
```

Comment:
because line no 83 to 86

https://github.com/mitodl/micromasters/pull/546/files#diff-30ff6c0b423aeb83e6ee53260122cac7R83

If we dont use this then instead of 404 it throws permission denied exception.